packages/flutter/lib/src/rendering/object.dart

```diff
@@ -183,6 +205,10 @@ class PaintingContext extends ClipContext {
     if (child.isRepaintBoundary) {
       stopRecordingIfNeeded();
       _compositeChild(child, offset);
+    } else if (child._wasRepaintBoundary) {
```

Comment:
here is where we automatically dispose layers for old repaint boundaries

Reply:
A future reader would probably appreciate a comment in the code stating that.

Reply:
done